src/webapp/components/settings-form/SettingsForm.tsx

```diff
@@ -0,0 +1,357 @@
+import React from "react";
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogTitle,
+    LinearProgress,
+    Paper,
+    Tab,
+    Tabs,
+    TextField,
+} from "@material-ui/core";
+
+import i18n from "$/utils/i18n";
+import { ModalSelector } from "$/webapp/components/modal-selector/ModalSelector";
+import { Maybe } from "$/utils/ts-utils";
+import { Dropdown, useSnackbar } from "@eyeseetea/d2-ui-components";
+import styled from "styled-components";
+import { getDaysPerMonthYear, getMonths, getUnits } from "$/utils/date";
+import { useAppContext } from "$/webapp/contexts/app-context";
+import { AppSettings } from "$/domain/entities/AppSettings";
+import { AppSettingsData } from "$/domain/entities/AppSettingsData";
+
+type SettingsFormProps = { onClose: () => void };
+
+export const SettingsForm = React.memo((props: SettingsFormProps) => {
+    const { onClose } = props;
+    const { compositionRoot, currentUser } = useAppContext();
+    const [settingsData, setSettingsData] = React.useState<AppSettingsData>();
+    const [appSettings, setAppSettings] = React.useState<AppSettings>();
+    const [selectedTab, setSelectedTab] = React.useState(0);
+    const [loading, setLoading] = React.useState(false);
+    const snackbar = useSnackbar();
+
+    React.useEffect(() => {
+        return compositionRoot.appSettings.get
+            .execute(currentUser)
+            .run(setAppSettings, console.error);
+    }, [compositionRoot.appSettings.get, currentUser]);
+
+    React.useEffect(() => {
+        return compositionRoot.appSettings.getData
+            .execute(currentUser)
+            .run(setSettingsData, console.error);
+    }, [compositionRoot.appSettings.getData, currentUser]);
+
+    const changeTab = (value: number) => {
+        setSelectedTab(value);
+    };
+
+    const updateSettings = (settings: AppSettings) => {
+        setAppSettings(settings);
+    };
+
+    const saveSettings = () => {
+        if (!appSettings) return;
+        setLoading(true);
+        compositionRoot.appSettings.save.execute(appSettings, currentUser).run(
+            () => {
+                snackbar.success(i18n.t("Settings saved"));
+                onClose();
+                setLoading(false);
+            },
+            () => {
+                setLoading(false);
+            }
+        );
+    };
```

Comment:
we have useCallbackEffect in other projects for callbacks which fire a side-effect.